cmd/kubeadm/app/phases/kubelet/config.go

```diff
@@ -82,6 +79,17 @@ func CreateConfigMap(cfg *kubeadmapi.ClusterConfiguration, client clientset.Inte
 		return err
 	}
 
+	// Apply patches to the KubeletConfiguration
+	if len(patchesDir) != 0 {
+		kubeletBytes, err = applyKubeletConfigPatches(kubeletBytes, patchesDir, os.Stdout)
```

Comment:
WriteConfigToDisk will apply once and here applied the second time. Can `kubeletBytes` be reused after applying?

Overall LGTM and this is just for duplicate logic between `CreateConfigMap()` and  `WriteConfigToDisk()`.

Reply:
>Can kubeletBytes be reused after applying?

Yes, when doing cluster upgrade, we read the kubelet configuration from `kube-system/kubelet-config` ConfigMap.

>Overall LGTM and this is just for duplicate logic between CreateConfigMap() and WriteConfigToDisk()

I'll consider merging these duplicate logics with one function.

Reply:
Got it.